ui/src/app/views/workspace-share/component.ts

```diff
@@ -61,32 +69,72 @@ export class WorkspaceShareComponent implements OnInit {
     }
   }
 
+  convertToEmail(username: string): string {
+    return username + '@fake-research-aou.org';
+  }
+
+  inputChange(): void {
+    this.userNotFound = false;
+  }
+
   addCollaborator(): void {
-    this.workspace.userRoles.push({email: this.toShare, role: this.accessLevel});
-    this.workspacesService.shareWorkspace(
-      this.workspace.namespace, this.workspace.id, {
-        workspaceEtag: this.workspace.etag,
-        items: this.workspace.userRoles
-      }).subscribe((resp: ShareWorkspaceResponse) => {
-        this.workspace.etag = resp.workspaceEtag;
+    if (this.inputNotInitialized) {
+      this.inputNotInitialized = false;
+      this.input.nativeElement.addEventListener('keydown', () => {
+        this.userNotFound = false;
```

Comment:
```
<input (keydown)="userNotFound = false" /* attrs... */ >
```

Allows you to dispense with the initialization flag and is easier to scan (to my mind).